cli/src/pcluster/validators/directory_service_validators.py

```diff
@@ -82,37 +82,20 @@ def _validate(self, password_secret_arn: str, region: str):
             for retro-compatibility.
         """
         try:
-            # We only require the secret to exist; we do not validate its content.
-            arn_components = password_secret_arn.split(":")
-            service = arn_components[2]
-            resource = arn_components[5]
+            service, resource = _get_service_and_resource(password_secret_arn)
             if service == "ssm":
-                resource = arn_components[5].split("/")[0]
-
+                resource = password_secret_arn.split(":")[5].split("/")[0]
             if service == "secretsmanager" and resource == "secret":
                 AWSApi.instance().secretsmanager.describe_secret(password_secret_arn)
             elif service == "ssm" and resource == "parameter" and region == "us-isob-east-1":
-                parameter_name = arn_components[5].split("/")[1]
+                parameter_name = password_secret_arn.split(":")[5].split("/")[1]
                 AWSApi.instance().ssm.get_parameter(parameter_name)
             else:
                 self._add_failure(
                     f"The secret {password_secret_arn} is not supported in region {region}.", FailureLevel.ERROR
                 )
         except AWSClientError as e:
-            if e.error_code in ("ResourceNotFoundExceptionSecrets", "ParameterNotFound"):
-                self._add_failure(f"The secret {password_secret_arn} does not exist.", FailureLevel.ERROR)
-            elif e.error_code == "AccessDeniedException":
-                self._add_failure(
-                    f"Cannot validate secret {password_secret_arn} due to lack of permissions. "
-                    "Please refer to ParallelCluster official documentation for more information.",
-                    FailureLevel.WARNING,
-                )
-            else:
-                self._add_failure(
-                    f"Cannot validate secret {password_secret_arn}. "
-                    "Please refer to ParallelCluster official documentation for more information.",
-                    FailureLevel.WARNING,
-                )
+            _handle_arn_aws_client_error(e, password_secret_arn, self)
```

Comment:
This should not be moved outside of this class into an external function, because this really depends on the type of API call that was performed in the caller method.

If you really want to avoid code duplication across multiple validators, the correct way is to have them as sibling classes (children of the same class, possibly an abstract one).

I would say, let's duplicate this code for the time being  (I have already created an internal task to refactor the various Secret validators we have in our code).

Reply:
> have them as sibling classes (children of the same class, possibly an abstract one).

I tried, that's the way I implemented it before. But doesn't work. I use a father class inherit from `Validator` class. But because there's a unit test called `test_slurm_all_validators_are_called` and it consider this father class as a validator that should be called. So that this unit test will fail.

Reply:
> This should not be moved outside of this class into an external function, because this really depends on the type of API call that was performed in the caller method.
But for now, it's the same as the newly added validator `MungeKeySecretSizeAndBase64Validator`. I didn't change any logic in this `PasswordSecretArnValidator`. The same as before.

Reply:
> But because there's a unit test called test_slurm_all_validators_are_called and it consider this father class as a validator that should be called. So that this unit test will fail.

There should be a way to exclude a validator class from that test.

Anyway, it's just a matter of refactoring this code into something more OOP-friendly. I will create a nice-to-have task for this.